src/ui/rcb-ui.svelte

```diff
@@ -1,317 +1,323 @@
+    import StateNeedsAuthInfo from "./states/state-needs-auth-info.svelte";
+    import ConditionalFullScreen from "./conditional-full-screen.svelte";
+    import Shell from "./shell.svelte";
+    import {type SubscribeResponse} from "../networking/responses/subscribe-response";
+    import {type BrandingInfoResponse} from "../networking/responses/branding-response";
+    import {
+        PurchaseFlowError,
+        PurchaseFlowErrorCode,
+        PurchaseOperationHelper,
+    } from "../helpers/purchase-operation-helper";
+    import {Backend} from "../networking/backend";
+    import ModalHeader from "./modal-header.svelte";
+    import IconCart from "./assets/icon-cart.svelte";
+    import BrandingInfoUI from "./branding-info-ui.svelte";
+    import SandboxBanner from "./sandbox-banner.svelte";
+    import {Colors} from "../assets/colors";
 
-  export let asModal = true;
-  export let customerEmail: string | undefined;
-  export let appUserId: string;
-  export let rcPackage: Package;
-  export let onFinished: () => void;
-  export let onError: (error: PurchaseFlowError) => void;
-  export let onClose: () => void;
-  export let purchases: Purchases;
-  export let backend: Backend;
-  export let purchaseOperationHelper: PurchaseOperationHelper;
+    export let asModal = true;
+    export let customerEmail: string | undefined;
+    export let appUserId: string;
+    export let rcPackage: Package;
+    export let subscriptionPurchaseOptionId: string | undefined;
+    export let onFinished: () => void;
+    export let onError: (error: PurchaseFlowError) => void;
+    export let onClose: () => void;
+    export let purchases: Purchases;
+    export let backend: Backend;
+    export let purchaseOperationHelper: PurchaseOperationHelper;
 
-  const colorVariables = Object.entries(Colors)
-    .map(([key, value]) => `--rc-color-${key}: ${value}`)
-    .join("; ");
+    const colorVariables = Object.entries(Colors)
+        .map(([key, value]) => `--rc-color-${key}: ${value}`)
+        .join("; ");
 
-  let productDetails: any = null;
-  let brandingInfo: BrandingInfoResponse | null = null;
-  let paymentInfoCollectionMetadata: SubscribeResponse | null = null;
-  let lastError: PurchaseFlowError | null = null;
-  const productId = rcPackage.rcBillingProduct?.identifier ?? null;
+    let productDetails: any = null;
+    let brandingInfo: BrandingInfoResponse | null = null;
+    let paymentInfoCollectionMetadata: SubscribeResponse | null = null;
+    let lastError: PurchaseFlowError | null = null;
+    const productId = rcPackage?.rcBillingProduct?.identifier ?? null;
+    const defaultPurchaseOptionId = rcPackage?.rcBillingProduct?.defaultSubscriptionPurchaseOptionId;
+    const optionId = subscriptionPurchaseOptionId ? subscriptionPurchaseOptionId : defaultPurchaseOptionId;
+    const purchaseOption = optionId ?
+        rcPackage?.rcBillingProduct?.subscriptionPurchaseOptions[optionId] : undefined;
```

Comment:
We should probably log something and fail in the case there is no purchase option I think.